alacritty/src/display/damage.rs

```diff
@@ -0,0 +1,345 @@
+use std::cmp;
+
+use glutin::Rect;
+
+use alacritty_terminal::selection::SelectionRange;
+use alacritty_terminal::term::{DamagedLine, RenderableCursor, SizeInfo, TermDamage};
+use alacritty_terminal::vi_mode::ViModeCursor;
+
+/// Minimum part of damaged square when intersecting damaged rectangles.
+pub const MIN_DAMAGED_SQUARE_PART: f32 = 0.4;
+
+/// An error when we for look ahead in iterator, used to try to collect rectangles which pretty
+/// close to each other, but don't fit into `MIN_DAMAGED_SQUARE_PART` just by looking on the first
+/// next damaged rectangle.
+pub const MIN_DAMAGED_SQUARE_PART_ERR: f32 = MIN_DAMAGED_SQUARE_PART / 2.;
+
+pub struct RendererDamage<'a> {
+    index: usize,
+    rect: Option<Rect>,
```

Comment:
What is this exactly, the total damage rectangle, so all other damage combined? Just `rect` doesn't really tell me much.

Reply:
This is the rectangle we'll either yeild `fn next` or its merged result with some other rect. So basically the one we're currently using for merging.

Reply:
There might be a better name then. Will have to look at the iterator method to really understand though probably.